src/player.h

```diff
@@ -1037,6 +1037,10 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
          * Check whether the player has a gun that uses the given type of ammo.
          */
         bool has_gun_for_ammo( const ammotype &at ) const;
+        bool has_magazine_for_ammo( const ammotype &at ) const;
+        bool has_magazine_for_gun( const item &gun ) const;
```

Comment:
You should use `Character::find_ammo` with `empty = true` and `radius = -1` to search for magazines and/or ammo. This function is aware of ammo pouches, bandoliers etc and as a result you won't need to later update your implementation.